mrjob/spark/runner.py

```diff
@@ -160,7 +168,14 @@ def _pick_spark_tmp_dir(self):
                     fully_qualify_hdfs_path('tmp/mrjob'), self._job_key)
 
     def _default_step_output_dir(self):
-        return posixpath.join(self._spark_tmp_dir, 'step-output')
+        return self.fs.join(self._spark_tmp_dir, 'step-output')
+
+    def _counter_output_dir(self, step_num):
+        return self.fs.join(
+            self._spark_tmp_dir, 'counter-output-step-%d' % step_num)
+
+    def counters(self):
+        return deepcopy(self._counters)
```

Comment:
Kind of curious why this returning a copy of counters, I am guessing you want to avoid the number being changed accidentally by client?

Reply:
Yeah. I mean, it can't be that important or I would have changed it ages ago, but I definitely don't mean for `self._counters` to be editable.